### ISSUE TITLE
sound generation should be in the view (right?)

### ISSUE THREAD

**Issue by jbphet** [phetsims/john-travoltage#93]
I am currently working on the sonfication of this simulation (see #92) and am doing most of the work in the view.  The sim currently generates sounds from the model.  This seems wrong to me - sonic expression of what is happening in the model strikes me as parallel to visual expression of what's happening in the model, so sounds should come from the view.

The dev team should decide if this is the convention we should follow and be consistent about it in future sims and, if we agree, the sound generation in this sim that is currently done in the model should be moved.

**Comment by samreid** [comment]
Yes, the audio should be treated as a view and not implemented in the model.  I also recommend that for each simulation screen we create a file named like SimulationScreenAudio.js or SimulationScreenSonification.js which is instantiated by the SimulationScreenView.js, instead of adding audio code into the SimulationScreenView.js file.  My only hesitation about this is if there is *a lot* of code in the view for events that are salient for audio events, and we want to piggyback off of it to avoid duplication, but my sense is this is not the case.

**Comment by jbphet** [comment]
I was thinking along similar lines - basically looking at the sonification as an audio version of the view in the Model View Controller sense, so treating it separately, and it would have its own copy of the model and would monitor the model to portray its various sounds.  In an ideal world, it would be created as a separate "view", but our framework doesn't really support that, which I'm guessing is why @samreid suggested having the SimulationScreenView create it.

**Comment by samreid** [comment]
> In an ideal world, it would be created as a separate "view", but our framework doesn't really support that, which I'm guessing is why @samreid suggested having the SimulationScreenView create it.

Yes, exactly.

> it would have its own copy of the model and would monitor the model to portray its various sounds

I presume you mean reference to the model.

**Comment by jbphet** [comment]
Yes indeed, it would have a reference to the model.

**Comment by samreid** [comment]
Question for developer meeting:

Should we redesign Screen.js so that audio can be created as a top level view?  Or just create the audio instance from the ScreenView?

**Comment by jessegreenberg** [comment]
Are all audio instances created in the ScreenView at the moment?  Or are some distributed to other ScreenView child nodes?  For instance does AppendageNode instrument its own sonification?  If individual nodes need their own audio would a single top level instance be sufficient, whether it is in the Screen or ScreenView?

**Comment by samreid** [comment]
> If individual nodes need their own audio would a single top level instance be sufficient, whether it is in the Screen or ScreenView?

I was envisioning an audio tree (parallel to how we currently have a visual tree), not everything necessarily collapsed into a single file.

But that raises a good point.  What if there is a sound effect that plays when you press a particular button?   Should the audio be implemented in the button callback (for simplicity?) Or should the button trigger some model emitter which the audio view can listen to (for separation of concerns)?  If we go with the latter, will there be a proliferation of model emitters that correspond to "UI stuff"?

**Comment by samreid** [comment]
Even though there is not specific support for it in Screen.js, Screen.initializeModelAndView could be overriden to initialize custom views.

**Comment by pixelzoom** [comment]
+1 for sonification in the view. I don't have time to comment on the specifics of how that might be done.

**Comment by jbphet** [comment]
In response to @jessegreenberg's comment above (https://github.com/phetsims/john-travoltage/issues/93#issuecomment-227185047), there currently is some sound generation taking place in the model itself.  This was done in the original implementation, and when I saw it I decided NOT to follow that convention for the sonification work, and logged this issue to track it.  As for whether the individual nodes or the larger view is responsible for the rest of the audio, it's a bit of a mix at this point, which is another reason why it would be good for us to have some agreed upon philosophy about how to approach this.
